src/swupd/manifest_test.go

```diff
@@ -254,3 +258,90 @@ func TestReadManifestFromFileEmpty(t *testing.T) {
 		t.Error("ReadManifestFromFile did not raise error on empty file")
 	}
 }
+
+func compareFiles(file1path string, file2path string) (bool, error) {
+	chunkSize := 64000
+	f1, err := os.Open(file1path)
+	if err != nil {
+		return false, err
+	}
+
+	f2, err := os.Open(file2path)
+	if err != nil {
+		return false, err
+	}
+
+	for {
+		b1 := make([]byte, chunkSize)
+		_, err1 := f1.Read(b1)
```

Comment:
Throwing away the number of bytes read is just wrong. Of course you probably don't want to invent co-routines, so you get into issues where read could return a less than full buffer... Probably best to avoid being clever here and just use bufio.